core/src/main/scala/kafka/tools/GetOffsetShell.scala

```diff
@@ -189,22 +209,26 @@ object GetOffsetShell {
     val topicFilter = IncludeList(group(1).getOrElse(".*"))
     val partitionFilter = group(2).map(_.toInt) match {
       case Some(partition) =>
-        (p: Int) => p == partition
+        UniquePartitionFilter(partition)
       case None =>
         val lowerRange = group(3).map(_.toInt).getOrElse(0)
         val upperRange = group(4).map(_.toInt).getOrElse(Int.MaxValue)
-        (p: Int) => p >= lowerRange && p < upperRange
+        PartitionRangeFilter(lowerRange, upperRange)
     }
-
-    tp => topicFilter.isTopicAllowed(tp.topic, excludeInternalTopics) && partitionFilter(tp.partition)
+    TopicFilterAndPartitionFilter(
+      topicFilter,
+      partitionFilter
+    )
   }
 
-  /**
```

Comment:
Did we mean to remove this comment?

Reply:
Oh no, this was a misoperate, this method was removed first and brought back since we need a test case for it. I have changed it back.